python/ray/autoscaler/aws/example-full.yaml

```diff
@@ -57,38 +53,45 @@ auth:
 # configurations below.
 #    ssh_private_key: /path/to/your/key.pem
 
-# Provider-specific config for the head node, e.g. instance type. By default
-# Ray will auto-configure unspecified fields such as SubnetId and KeyName.
-# For more documentation on available fields, see:
-# http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.ServiceResource.create_instances
-head_node:
-    InstanceType: m5.large
-    ImageId: ami-0a2363a9cff180a64 # Deep Learning AMI (Ubuntu) Version 30
-
-    # You can provision additional disk space with a conf as follows
-    BlockDeviceMappings:
-        - DeviceName: /dev/sda1
-          Ebs:
-              VolumeSize: 100
-
-    # Additional options in the boto docs.
-
-# Provider-specific config for worker nodes, e.g. instance type. By default
-# Ray will auto-configure unspecified fields such as SubnetId and KeyName.
-# For more documentation on available fields, see:
-# http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.ServiceResource.create_instances
-worker_nodes:
-    InstanceType: m5.large
-    ImageId: ami-0a2363a9cff180a64 # Deep Learning AMI (Ubuntu) Version 30
-
-    # Run workers on spot by default. Comment this out to use on-demand.
-    InstanceMarketOptions:
-        MarketType: spot
-        # Additional options can be found in the boto docs, e.g.
-        #   SpotOptions:
-        #       MaxPrice: MAX_HOURLY_PRICE
-
-    # Additional options in the boto docs.
```

Comment:
```
head_node: {}
worker_nodes: {}
```

Reply:
Hmm, I'm a little confused -- it seems that `bootstrap_aws` fills in all sorts of interesting information into the  `head_node` and `worker_nodes` but then the rest of cluster launching/autoscaling process doesn't use this info at all 
(it just draws from the node-type info).
(Correct me if I'm missing something.) 

So... resurfacing this question: should we consider updating all of the logic bootstrap_aws to be multi-node-compatible before making these config changes?

Reply:
hmm... that can't be right...

Reply:
ok, sry -- I see NodeLauncher and get_or_create_head_node do the merging